scripts/travis/external_build_printlog.sh

```diff
@@ -44,7 +45,19 @@ while [ $SECONDS -lt $end ]; do
 
     if [ $SECONDS -gt $minute_end ]; then
         minute_end=$((SECONDS+60))
-        echo "Still waiting for build to complete..."
+        echo "Still waiting for build to complete $(((SECONDS-LOG_WAITING_START)/60))m:$(((SECONDS-LOG_WAITING_START)%60))s..."
+    fi
+
+    if [ "${TRAVIS}" = "true" ]; then
+        # under travis, if we have passed the 1h:45m mark, the build is going to likely fail due to timeout.
+        # instead of failing, we want to exit the build with success, indicating where the log could be retrieved later on.
+        # ( note that there migth be an issue with that build, but we don't want to cap it via travis timeouts. )
+        if [ $((SECONDS-LOG_WAITING_START)) -gt $((60*105)) ]; then
+            echo "Build is taking too long. Travis is going to timeout this build, so we'll tell travis that we're done for now."
+            echo "Once this build is complete, you could get a complete log by typing:"
+            echo "./scripts/travis/external_build_printlog.sh \"${BUILD_LOG_PATH}\" \"${BUILD_COMPLETE_PATH}\" ${NO_SIGN_REQUEST}"
+            exit 0
+        fi
```

Comment:
This is counterintuitive. Timeout == something wrong => fail

Reply:
True, but unfortunately, this is the success case. I would have increased the timeout, but travis doesn't have this feature. 
If the build is going to take 3.5 hours, I don't want travis to tell me that the build is broken all the time ( that won't help me. Instead, printing this message would.. )